lib/ex_doc.ex

```diff
@@ -93,20 +93,43 @@ defmodule ExDoc do
     find_formatter(config.formatter).run(docs, config)
   end
 
+  @doc false
   # Builds configuration by merging `options`, and normalizing the options.
   @spec build_config(String.t, String.t, Keyword.t) :: ExDoc.Config.t
-  defp build_config(project, vsn, options) do
+  def build_config(project, vsn, options) do
     options = normalize_options(options)
+
     preconfig = %Config{
       project: project,
       version: vsn,
-      main: options[:main],
-      homepage_url: options[:homepage_url],
       source_root: options[:source_root] || File.cwd!,
     }
     struct(preconfig, options)
   end
 
+  @spec normalize_options(keyword) :: keyword
+  def normalize_options(options) do
+    options = normalize_options(options, :source_ref, ExDoc.Config.default(:source_ref))
+
+    options
```

Comment:
nope. this was done intetionally.
You cannot do that, because in the last line when I access `options[:source_ref]` I need the updated version that `normalize_options($options, :source_ref, ExDoc.Config.default(:source_ref))` updates